### ISSUE TITLE
[MJAVADOC-538] JVM "JAVA_TOOL_OPTIONS" or "_JAVA_OPTIONS" message detected as javadoc warning, triggers failure 

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-javadoc-plugin#807]
**[Yoann Rodière](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=yrodiere)** opened **[MJAVADOC-538](https://issues.apache.org/jira/browse/MJAVADOC-538?redirect=false)** and commented

When an environment variable `_JAVA_OPTIONS` or `JAVA_TOOL_OPTIONS` is defined, every JVM that starts will output a message to inform users of these options being picked up:

```
$ export JAVA_TOOL_OPTIONS=-Dfoo
$ java -version                 
Picked up JAVA_TOOL_OPTIONS: -Dfoo
java version "1.8.0_181"
Java(TM) SE Runtime Environment (build 1.8.0_181-b13)
Java HotSpot(TM) 64-Bit Server VM (build 25.181-b13, mixed mode)
```

Apparently this message cannot be suppressed, that's on purpose an won't change: https://bugs.openjdk.java.net/browse/JDK-8039152

 

Unfortunately, this message seems to be interpreted by the maven-javadoc-plugin as a warning:

```
[INFO] --- maven-javadoc-plugin:3.0.1:javadoc-no-fork (generate-javadoc) @ hibernate-search-util-internal-common ---
[INFO] The goal 'org.apache.maven.plugins:maven-javadoc-plugin:3.0.1:javadoc' has not been previously called for the module: 'org.hibernate.search:hibernate-search-util-internal-test:jar:6.0.0-SNAPSHOT'. Trying to invoke it...
[WARN] Maven will be executed in interactive mode, but no input stream has been configured for this MavenInvoker instance.
Picked up JAVA_TOOL_OPTIONS: -Dfoo
[WARNING] Creating fake javadoc directory to prevent repeated invocations: /home/yrodiere/workspaces/main/hibernate-search-parent/util/internal/test/target/site/apidocs
[ERROR] Error fetching link: /home/yrodiere/workspaces/main/hibernate-search-parent/util/internal/test/target/site/apidocs/package-list. Ignored it.
[WARNING] Javadoc Warnings
[WARNING] Picked up JAVA_TOOL_OPTIONS: -Dfoo
```

(see the last line, the message appears in the list of warnings)

No big deal... until you enable the [failOnWarnings](https://maven.apache.org/plugins/maven-javadoc-plugin/javadoc-no-fork-mojo.html#failOnWarnings) option. Then the whole build will fail just because some JVM detected an environment variable...

I couldn't find any workaround, except disabling the `failOnWarnings` options, but that's more a surrender than a workaround.

Note that the definition of `_JAVA_OPTIONS` or `JAVA_TOOL_OPTIONS` is not exactly an exotic use case, especially on Continuous Integration platforms. [Travis CI uses `_JAVA_OPTIONS`](https://docs.travis-ci.com/user/build-environment-updates/2017-09-06/#added) to set JVM memory limits in its containerized environments, and [the Jenkins CI Pipeline Maven plugin uses `JAVA_TOOL_OPTIONS`](https://wiki.jenkins.io/display/JENKINS/Pipeline+Maven+Plugin#PipelineMavenPlugin-WhydoIseemessages%22[WARNING)PickedupJAVA_TOOL_OPTIONS...%22inthebuildlogs?] to pass options to Maven processes transparently.

---

**Affects:** 3.0.1

**Issue Links:**
- [MJAVADOC-541](https://issues.apache.org/jira/browse/MJAVADOC-541) Jenkinsfile withMaven causing JAVA_TOOL_OPTIONS message to be picked up as a Javadoc warning
 (_**"is duplicated by"**_)

**Remote Links:**
- [GitHub Pull Request #11
](https://github.com/apache/maven-javadoc-plugin/pull/11)

1 votes, 5 watchers

**Comment by jira-importer** [comment]
**[Kevin Risden](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=risdenk)** commented

Created a pull request: https://github.com/apache/maven-javadoc-plugin/pull/11

**Comment by jira-importer** [comment]
**[Robert Scholte](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=rfscholte)** commented

Fixed in [631a84ee0a54acc916b0958b9c5b86a5d417712d](https://gitbox.apache.org/repos/asf?p=maven-javadoc-plugin.git;a=commit;h=631a84ee0a54acc916b0958b9c5b86a5d417712d)
Thanks for the patch!
